### PR TITLE
[test reporting] Add support for combining a set of test results

### PR DESCRIPTION
**Pull request by daall**
Signed-off-by: Danny Allen <daall@microsoft.com>

### Description of PR

Summary: [test reporting] Add support for combining a set of test results
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Running the tests in individual mode produces a huge directory of `*.xml` test result files. These need to be combined together in order to get a comprehensive view of the results for a given test run.

#### How did you do it?
I updated the parser logic to support a full directory of test results, with the following restrictions:
- The metadata should match across all files (i.e. we assume that the testbed/hwsku/version/etc. are not going to be changing throughout a single test run)
- The summary + test cases are simply combined together.

There were also some changes to the parser logic:
- Some files may not have test results (e.g. pretest + posttest) if checks are not run, so the requirement that every file must have test cases has been removed.
- Skipped messages are now included in the json output
- "Markers" is no longer required.

#### How did you verify/test it?
Added a new test case + updated the old ones w/ the latest changes. Also ran the CLI in `-d` mode against a few nightly test runs and they were combined as expected.

#### Any platform specific information?
N/A

#### Supported testbed topology if it's a new test case?
N/A

### CODE REVIEW

File: test_reporting/junit_xml_parser.py [hunk 1]
```diff
@@ -122,36 +128,91 @@ def validate_junit_xml_file(document_name):
...
+    metadata = {}
+    doc_list = glob.glob(os.path.join(directory_name, "*.xml"))
+    doc_list += glob.glob(os.path.join(directory_name, "**", "*.xml"))
+
+    total_size = 0
+    for document in doc_list:
+        total_size += os.path.getsize(document)
+
+    if total_size > MAXIMUM_XML_SIZE:
+        raise JUnitXMLValidationError("provided directory is too large")
```
Comment (wangxin) [inline test_reporting/junit_xml_parser.py:151]: Total size is limited to 10MB, could this be too strict? The junit xml may contain captured log and have floated size.

**Review by wangxin — APPROVED** [review]
LGTM except a minor concern.

File: test_reporting/junit_xml_parser.py (hunk 1 above)
Comment (daall) [reply test_reporting/junit_xml_parser.py:166]: I think that's a pretty valid concern. I've bumped it up to 20MB which should be plenty. I think the largest individual file I've seen is ~250kb.

**Comment by daall** [comment]
retest vsimage please

**Pull request merged**